### DISCUSSION TITLE
Server request from client side don't get limited.

### DISCUSSION THREAD

**Discussion by KirillGoorevich** [nfriedly/express-rate-limit#315] [Q&A]
I have a react app with server and client side, say I set in 15 minutes max 10 requests when I open the server in browser and refresh 10 times I get the limit exceeded message. However if I go to the client side and make server api calls from the the client, for instance a login attempt, I don't get any limitation, that is to say a hacker that knows a username can brute force a password without anything to limit him.
why is it that express-rate-limit only works if I reload the server on a webpage.
are there parameters that express-react-limit use that don't get passed when a client side calls the server api? Like the IP that made the request?
client side, login:

```js
doSubmit = async () => {
    try {
      const user = { ...this.state.data };
      await login(user);
      toast.success("you logged successfully");
      window.location = "/";
    } catch (error) {
      if (error.response && error.response.status === 400)
        this.setState({
          errors: { email: "Invalid email or password" },
        });
    }
```
    
client side,login service:

```js
export const login = async user => {
  const {
    data: { token },
  } = await http.post(`${apiUrl}/users/login`, user);
  localStorage.setItem("token", token);
};
```

server side login api:

```js
router.post("/login", async (req, res) => {
  const { error } = validateSignin(req.body);
  if (error) {
    console.log(chalk.redBright(error.details[0].message));
    return res.status(400).send(error.details[0].message);
  }

  let user = await User.findOne({ email: req.body.email });
  if (!user) {
    console.log(chalk.redBright("Invalid email"));
    return res.status(400).send("Invalid email or password.");
  }

  const validPassword = comparePassword(req.body.password, user.password);
  if (!validPassword) {
    console.log(chalk.redBright("Invalid password"));
    return res.status(400).send("Invalid email or password.");
  }

  res.json({
    token: generateAuthToken(user),
  });
});
```

**Comment by nfriedly** [comment]
It doesn't look like `login()` actually checks the response. `http.post()` could have some status checking that isn't shown here, but based on your description, I don't think it does. I think you're setting the localStorage and showing the "you logged successfully" toast even when rate-limited.

You should try logging the response before doing the de-structuring in `login()` and/or see if token is actually getting set properly in localStorage when rate-limited.

**Comment by KirillGoorevich** [comment]
I was using the library in app.js, tried using it in api.js and worked I guess, app.js was unaware of what was going on in api.js even though it was required.

**Comment by KirillGoorevich** [comment]
thank you for your reply, as i stated before I did solve the problem,
as for the check in http.post I do have:
if (expectedError) toast.error(`An error occurred: ${error.message}`);
